include/rapidcheck/state/Commands.hpp

```diff
@@ -1,9 +1,109 @@
 #pragma once
 
 #include "rapidcheck/detail/Results.h"
+#include "rapidcheck/state/gen/ParallelCommands.hpp"
+#include "rapidcheck/Assertions.h"
+#include <thread>
+#include <mutex>
+#include "rapidcheck/detail/Barrier.h"
 
 namespace rc {
 namespace state {
+namespace detail {
+
+
+template <typename Model, typename Cmd>
+struct CommandResult
+{
+  CommandResult(const std::shared_ptr<const Cmd>& command,
+                std::function<void(const Model&)>&& verifyFunc)
+    : command(command)
+    , verifyFunc(std::move(verifyFunc))
+  {
+  }
+
+  std::shared_ptr<const Cmd> command;
```

Comment:
Hmmm...I'm not 100% sure what you are after so this answer may not make any sense :) 
This class exist since the verify function needs to be associated with the corresponding command somehow. This could e.g. be solved by using a hash map but that won't be as efficient as keeping them in the same object.